test/e2e/prometheus_test.go

```diff
@@ -5495,6 +5495,280 @@ func testPrometheusReconciliationOnSecretChanges(t *testing.T) {
 	require.NoError(t, err)
 }
 
+func verifyPrometheusConfig(ns, promName, expectedConfig string) (bool, error) {
+	pods, err := framework.KubeClient.CoreV1().Pods(ns).List(context.Background(), metav1.ListOptions{
+		LabelSelector: fmt.Sprintf("prometheus=%s", promName),
+	})
+	if err != nil {
+		return false, err
+	}
+
+	if len(pods.Items) == 0 {
+		return false, fmt.Errorf("no prometheus pods found")
+	}
+
+	pod := pods.Items[0]
+	stdout, _, err := framework.ExecWithOptions(context.Background(), testFramework.ExecOptions{
+		Command: []string{
+			"/bin/sh", "-c", "cat /etc/prometheus/config_out/prometheus.env.yaml",
+		},
+		Namespace:     ns,
+		PodName:       pod.Name,
+		ContainerName: "prometheus",
+		CaptureStdout: true,
+		CaptureStderr: true,
+		Stdin:         nil,
+	})
+	if err != nil {
+		return false, err
+	}
+
+	return strings.Contains(stdout, expectedConfig), nil
+}
+
+func testPrometheusUTF8LabelSupport(t *testing.T) {
+	skipPrometheusTests(t)
+	t.Parallel()
+
+	testCtx := framework.NewTestCtx(t)
+	defer testCtx.Cleanup(t)
+	ns := framework.CreateNamespace(context.Background(), t, testCtx)
+	framework.SetupPrometheusRBAC(context.Background(), t, testCtx, ns)
+
+	_, err := framework.CreateOrUpdatePrometheusOperator(
+		context.Background(),
+		ns,
+		[]string{ns},
+		nil,
+		[]string{ns},
+		nil,
+		false,
+		true,
+		true,
+	)
+	require.NoError(t, err)
+
+	t.Run("UTF8RelabelConfigWithPrometheus3", func(t *testing.T) {
+		testUTF8RelabelConfigWithPrometheus3(t, ns)
+	})
+
+	t.Run("UTF8RelabelConfigWithPrometheus2", func(t *testing.T) {
+		testUTF8RelabelConfigWithPrometheus2(t, ns)
+	})
+
+	t.Run("UTF8ServiceMonitorLabelsWithPrometheus3", func(t *testing.T) {
+		testUTF8ServiceMonitorLabelsWithPrometheus3(t, ns)
+	})
+}
+
+func testUTF8RelabelConfigWithPrometheus3(t *testing.T, ns string) {
+	name := "prometheus-utf8-relabel"
+
+	prom := framework.MakeBasicPrometheus(ns, name, "test-app", 1)
+	prom.Spec.Version = operator.DefaultPrometheusVersion
+
+	_, err := framework.CreatePrometheusAndWaitUntilReady(context.Background(), ns, prom)
```

Comment:
It's safer/faster to create the Prometheus resource last because it ensures that the generated config is up-to-date.

I would also recommend testing with a PrometheusRule resource.